<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Gene GRangesList to ggplot2 grob — gene2gg • splicejam</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Gene GRangesList to ggplot2 grob — gene2gg" />
<meta property="og:description" content="Gene GRangesList to ggplot2 grob" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">splicejam</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.67.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/basic-rnaseq-workflow.html">Splicejam basic RNAseq workflow</a>
    </li>
    <li>
      <a href="../articles/create-a-sashimi-plot.html">Create a Sashimi Plot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/jmw86069/splicejam/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene GRangesList to ggplot2 grob</h1>
    <small class="dont-index">Source: <a href='http://github.com/jmw86069/splicejam/blob/master/R/jambio-gg.R'><code>R/jambio-gg.R</code></a></small>
    <div class="hidden name"><code>gene2gg.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Gene GRangesList to ggplot2 grob</p>
    </div>

    <pre class="usage"><span class='fu'>gene2gg</span><span class='op'>(</span>
  gene <span class='op'>=</span> <span class='cn'>NULL</span>,
  tx <span class='op'>=</span> <span class='cn'>NULL</span>,
  flatExonsByGene <span class='op'>=</span> <span class='cn'>NULL</span>,
  flatExonsByTx <span class='op'>=</span> <span class='cn'>NULL</span>,
  geneColor <span class='op'>=</span> <span class='st'>"dodgerblue"</span>,
  labelExons <span class='op'>=</span> <span class='cn'>TRUE</span>,
  exonLabelAngle <span class='op'>=</span> <span class='fl'>90</span>,
  exonLabelSize <span class='op'>=</span> <span class='fl'>8</span>,
  geneSymbolColname <span class='op'>=</span> <span class='st'>"gene_name"</span>,
  newValues <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>feature_type <span class='op'>=</span> <span class='st'>"gap"</span>, subclass <span class='op'>=</span> <span class='st'>"gap"</span>, gene_nameExon <span class='op'>=</span> <span class='st'>"gap"</span><span class='op'>)</span>,
  gene_order <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"first"</span>, <span class='st'>"last"</span><span class='op'>)</span>,
  return_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"grob"</span>, <span class='st'>"df"</span><span class='op'>)</span>,
  ref2c <span class='op'>=</span> <span class='cn'>NULL</span>,
  hjust <span class='op'>=</span> <span class='op'>-</span><span class='fl'>0.2</span>,
  vjust <span class='op'>=</span> <span class='fl'>0.5</span>,
  direction <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"both"</span>, <span class='st'>"x"</span>, <span class='st'>"y"</span><span class='op'>)</span>,
  compressGaps <span class='op'>=</span> <span class='cn'>TRUE</span>,
  tx2geneDF <span class='op'>=</span> <span class='cn'>NULL</span>,
  label_coords <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gene</th>
      <td><p>character string of the gene to plot, compared
with <code><a href='https://rdrr.io/r/base/names.html'>names(flatExonsByGene)</a></code> and <code>values(flatExonsByTx)$gene_name</code>.</p></td>
    </tr>
    <tr>
      <th>tx</th>
      <td><p>character vector of the transcripts to plot, useful
when specifying specific transcripts. Values are matched with
<code><a href='https://rdrr.io/r/base/names.html'>names(flatExonsByTx)</a></code>.</p></td>
    </tr>
    <tr>
      <th>flatExonsByGene, flatExonsByTx</th>
      <td><p>GRangesList objects, named
by <code>"gene_name"</code> or <code>"transcript_id"</code> respectively, containing
disjoint (non-overlapping) exons within each GRangesList
element. The data is expected to be in the form provided by
<code><a href='flattenExonsBy.html'>flattenExonsBy()</a></code>.</p></td>
    </tr>
    <tr>
      <th>geneColor</th>
      <td><p>character color used as the base color for
exons, where the color is varied for each feature type or
subclass.</p></td>
    </tr>
    <tr>
      <th>labelExons</th>
      <td><p>logical indicating whether to print text
labels beneath each exon, using the values in colname
<code>"gene_nameExon"</code>. Typically the gene and transcripts are
named using consistent names, in which case one exon label
is placed at the bottom of the lowest transcript for each
unique exon label.</p></td>
    </tr>
    <tr>
      <th>exonLabelAngle</th>
      <td><p>numeric angle in degrees (0 to 360)
indicating how to rotate exon labels, where <code>90</code> is
vertical, and <code>0</code> is horizontal.</p></td>
    </tr>
    <tr>
      <th>exonLabelSize</th>
      <td><p>numeric value or <code>unit</code> object from <code><a href='https://rdrr.io/r/grid/unit.html'>grid::unit()</a></code>.
Numeric values are assumed to have unit <code>"pt"</code> which refers to
font point size. Used to size exon labels when <code>labelExons=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>newValues</th>
      <td><p>argument passed to <code><a href='addGRLgaps.html'>addGRLgaps()</a></code> to fill
column values for newly created gap entries. It is useful
to have <code>feature_type="gap"</code> so gaps have a different value
than exons. It is also useful to have <code>subclass="gap"</code>
when there are <code>"cds"</code> and <code>"noncds"</code> entries in the
provided <code>flatExonsByGene</code> data.</p></td>
    </tr>
    <tr>
      <th>gene_order</th>
      <td><p>character value indicating whether the
flattened gene model should be plotted <code>"first"</code> above the
transcript exon models, or <code>"last"</code> and below the
transcript exon models.</p></td>
    </tr>
    <tr>
      <th>return_type</th>
      <td><p>character value indicating whether to return
the ggplot graphic object <code>"grob"</code>, or the data.frame
<code>"df"</code> used to create the ggplot object.</p></td>
    </tr>
    <tr>
      <th>ref2c</th>
      <td><p>list output from <code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code> which
contains among other things, the <code>trans_grc</code> data of
class <code>"trans"</code> used in <code><a href='https://ggplot2.tidyverse.org/reference/coord_trans.html'>ggplot2::coord_trans()</a></code> or
<code><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>ggplot2::scale_x_continuous()</a></code>.</p></td>
    </tr>
    <tr>
      <th>hjust, vjust</th>
      <td><p>numeric value to position exon labels
passed to <code><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>ggrepel::geom_text_repel()</a></code>.</p></td>
    </tr>
    <tr>
      <th>direction</th>
      <td><p>argument passed to <code><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>ggrepel::geom_text_repel()</a></code>
to restrict placement of labels to one axis direction.</p></td>
    </tr>
    <tr>
      <th>compressGaps</th>
      <td><p>logical indicating whether to compress gaps
between exons. When <code>ref2c</code> is supplied, this argument is
ignored and the supplied <code>ref2c</code> is used directly.</p></td>
    </tr>
    <tr>
      <th>tx2geneDF</th>
      <td><p>data.frame or NULL, optionally used to help
identify matching transcripts for the requested <code>gene</code> value,
used when <code>"gene_name"</code> is not present in <code>values(flatExonsByTx)</code>.</p></td>
    </tr>
    <tr>
      <th>label_coords</th>
      <td><p>numeric vector length 2, optional range of
genomic coordinates to restrict labels, so labels are not
arranged by <code><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>ggrepel::geom_text_repel()</a></code> even when <code>coord_cartesian()</code>
is used to zoom into a specific x-axis range.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical indicating whether to print verbose output.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments are passed to relevant functions
as needed, including <code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function is intended to help plot gene and transcript exon
models, and is a lightweight wrapper around <code><a href='grl2df.html'>grl2df()</a></code>.</p>
<p>It takes <code>flatExonsByGene</code> which is the output from
<code><a href='flattenExonsBy.html'>flattenExonsBy()</a></code>, and essentially plots the end result
for review.</p>
<p>Alternatively, when <code>return_type="df"</code>, the output is
the <code>data.frame</code> used to produce the ggplot, which allows
for more customization.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other jam plot functions: 
<code><a href='bgaPlotly3d.html'>bgaPlotly3d</a>()</code>,
<code><a href='factor2label.html'>factor2label</a>()</code>,
<code><a href='grl2df.html'>grl2df</a>()</code>,
<code><a href='jitter_norm.html'>jitter_norm</a>()</code>,
<code><a href='plotSashimi.html'>plotSashimi</a>()</code>,
<code><a href='prepareSashimi.html'>prepareSashimi</a>()</code>,
<code><a href='stackJunctions.html'>stackJunctions</a>()</code></p>
<p>Other jam ggplot2 functions: 
<code><a href='geom_diagonal_wide_arc.html'>geom_diagonal_wide_arc</a>()</code>,
<code><a href='plotSashimi.html'>plotSashimi</a>()</code>,
<code><a href='splicejam-extensions.html'>splicejam-extensions</a></code>,
<code><a href='to_basic.GeomShape.html'>to_basic.GeomShape</a>()</code></p>
<p>Other splicejam core functions: 
<code><a href='exoncov2polygon.html'>exoncov2polygon</a>()</code>,
<code><a href='grl2df.html'>grl2df</a>()</code>,
<code><a href='make_ref2compressed.html'>make_ref2compressed</a>()</code>,
<code><a href='plotSashimi.html'>plotSashimi</a>()</code>,
<code><a href='prepareSashimi.html'>prepareSashimi</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Assume we start with flattened gene exons</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_exon_wide_gr</span><span class='op'>)</span>;
<span class='va'>test_flatExonsByGene</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'>split</span><span class='op'>(</span><span class='va'>test_exon_wide_gr</span>,
   <span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'>values</span><span class='op'>(</span><span class='va'>test_exon_wide_gr</span><span class='op'>)</span><span class='op'>[</span>,<span class='st'>"gene_name"</span><span class='op'>]</span><span class='op'>)</span>;

<span class='co'># The most basic plot of exons</span>
<span class='fu'>gene2gg</span><span class='op'>(</span>gene<span class='op'>=</span><span class='st'>"TestGene1"</span>, flatExonsByGene<span class='op'>=</span><span class='va'>test_flatExonsByGene</span><span class='op'>)</span>;
</div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘ggplot2’ was built under R version 3.6.2</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘ggforce’ was built under R version 3.6.2</span></div><div class='output co'>#&gt; <span class='warning'>Warning: failed to set names on the unlisted CompressedRleList object</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘data.table’ was built under R version 3.6.2</span></div><div class='img'><img src='gene2gg-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># You can be fancy and number the exons</span>
<span class='va'>test_flatExonsByGene</span> <span class='op'>&lt;-</span> <span class='fu'><a href='assignGRLexonNames.html'>assignGRLexonNames</a></span><span class='op'>(</span><span class='va'>test_flatExonsByGene</span>,
   geneSymbolColname<span class='op'>=</span><span class='st'>"gene_name"</span><span class='op'>)</span>;
<span class='fu'>gene2gg</span><span class='op'>(</span>gene<span class='op'>=</span><span class='st'>"TestGene1"</span>, flatExonsByGene<span class='op'>=</span><span class='va'>test_flatExonsByGene</span><span class='op'>)</span>;
</div><div class='output co'>#&gt; <span class='warning'>Warning: failed to set names on the unlisted CompressedRleList object</span></div><div class='img'><img src='gene2gg-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># Or the exon labels can be hidden</span>
<span class='fu'>gene2gg</span><span class='op'>(</span>gene<span class='op'>=</span><span class='st'>"TestGene1"</span>, flatExonsByGene<span class='op'>=</span><span class='va'>test_flatExonsByGene</span>, labelExons<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: failed to set names on the unlisted CompressedRleList object</span></div><div class='img'><img src='gene2gg-3.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='kw'>if</span> <span class='op'>(</span><span class='fl'>1</span> <span class='op'>==</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>{</span>
   <span class='co'>## Do not run automated examples until sample data is available</span>

<span class='va'>ggGria1</span> <span class='op'>&lt;-</span> <span class='fu'>gene2gg</span><span class='op'>(</span><span class='st'>"Gria1"</span>,
   flatExonsByGene<span class='op'>=</span><span class='va'>flatExonsByGeneCds</span><span class='op'>)</span>;

<span class='co'>## if transcript exons are available</span>
<span class='va'>ggGria1</span> <span class='op'>&lt;-</span> <span class='fu'>gene2gg</span><span class='op'>(</span><span class='st'>"Gria1"</span>,
   flatExonsByGene<span class='op'>=</span><span class='va'>flatExonsByGene</span>,
   flatExonsByTx<span class='op'>=</span><span class='va'>flatExonsByTx</span><span class='op'>)</span>;

<span class='op'>}</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


