<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Jam Sashimi plot — plotSashimi • splicejam</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Jam Sashimi plot — plotSashimi" />
<meta property="og:description" content="Jam Sashimi plot" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">splicejam</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.63.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/basic-rnaseq-workflow.html">Splicejam basic RNAseq workflow</a>
    </li>
    <li>
      <a href="../articles/create-a-sashimi-plot.html">Create a Sashimi Plot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/jmw86069/splicejam/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Jam Sashimi plot</h1>
    <small class="dont-index">Source: <a href='http://github.com/jmw86069/splicejam/blob/master/R/jambio-gg.R'><code>R/jambio-gg.R</code></a></small>
    <div class="hidden name"><code>plotSashimi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Jam Sashimi plot</p>
    </div>

    <pre class="usage"><span class='fu'>plotSashimi</span><span class='op'>(</span>
  <span class='va'>sashimi</span>,
  show <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"coverage"</span>, <span class='st'>"junction"</span>, <span class='st'>"junctionLabels"</span><span class='op'>)</span>,
  coord_method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"scale"</span>, <span class='st'>"coord"</span>, <span class='st'>"none"</span><span class='op'>)</span>,
  exonsGrl <span class='op'>=</span> <span class='cn'>NULL</span>,
  junc_color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/jamba/man/alpha2col.html'>alpha2col</a></span><span class='op'>(</span><span class='st'>"goldenrod2"</span>, <span class='fl'>0.3</span><span class='op'>)</span>,
  junc_fill <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/jamba/man/alpha2col.html'>alpha2col</a></span><span class='op'>(</span><span class='st'>"goldenrod2"</span>, <span class='fl'>0.9</span><span class='op'>)</span>,
  junc_alpha <span class='op'>=</span> <span class='fl'>0.8</span>,
  junc_accuracy <span class='op'>=</span> <span class='fl'>1</span>,
  fill_scheme <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"sample_id"</span>, <span class='st'>"exon"</span><span class='op'>)</span>,
  color_sub <span class='op'>=</span> <span class='cn'>NULL</span>,
  ylabel <span class='op'>=</span> <span class='st'>"read depth"</span>,
  xlabel <span class='op'>=</span> <span class='cn'>NULL</span>,
  xlabel_ref <span class='op'>=</span> <span class='cn'>TRUE</span>,
  use_jam_themes <span class='op'>=</span> <span class='cn'>TRUE</span>,
  apply_facet <span class='op'>=</span> <span class='cn'>TRUE</span>,
  facet_scales <span class='op'>=</span> <span class='st'>"free_y"</span>,
  ref2c <span class='op'>=</span> <span class='cn'>NULL</span>,
  label_coords <span class='op'>=</span> <span class='cn'>NULL</span>,
  do_highlight <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sashimi</th>
      <td><p>Sashimi data prepared by <code><a href='prepareSashimi.html'>prepareSashimi()</a></code> which
is a <code>list</code> with <code>covDF</code> coverage data in data.frame format,
<code>juncDF</code> junction data in data.frame format,
<code>juncLabelDF</code> junction label coordinates in data.frame format,
<code>exonLabelDF</code> exon label coordinates per coverage polygon in
data.frame format,
<code>ref2c</code> list output from <code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code> to
transform genomic coordinates.</p></td>
    </tr>
    <tr>
      <th>show</th>
      <td><p>character vector of Sashimi plot features to include:
<code>"coverage"</code> sequence read coverage data;
<code>"junction"</code> splice junction read data.</p></td>
    </tr>
    <tr>
      <th>coord_method</th>
      <td><p>character value indicating the type of
coordinate scaling to use:
<code>"scale"</code> uses <code><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>ggplot2::scale_x_continuous()</a></code>;
<code>"coord"</code> uses <code><a href='https://ggplot2.tidyverse.org/reference/coord_trans.html'>ggplot2::coord_trans()</a></code>;
<code>"none"</code> does not compress genomic coordinates.</p></td>
    </tr>
    <tr>
      <th>exonsGrl</th>
      <td><p>GRangesList object with one or more gene or
transcript exon models, where exons are disjoint (not
overlapping.)</p></td>
    </tr>
    <tr>
      <th>junc_color, junc_fill</th>
      <td><p>character string with valid R color,
used for junction outline, and fill, for the junction arc
polygon. Alpha transparency is recommended for <code>junc_fill</code>
so overlapping junction arcs are visible.</p></td>
    </tr>
    <tr>
      <th>junc_alpha</th>
      <td><p>numeric value between 0 and 1, to define the
alpha transparency used for junction colors, where 0 is
fully transparent, and 1 is completely non-transparent.</p></td>
    </tr>
    <tr>
      <th>fill_scheme</th>
      <td><p>character string for how the exon coverages
will be color-filled: <code>"exon"</code> will define colors for each
distinct exon, using the GRanges names from <code>flatExonsByGene</code>;
<code>"sample_id"</code> to color all exons the same by sample_id.</p></td>
    </tr>
    <tr>
      <th>color_sub</th>
      <td><p>optional character vector of R compatible colors
or hex strings, whose names are used to color or fill features
in the ggplot object. For example, if <code>fill_sheme="sample_id"</code>
the <code>color_sub</code> should have names for each <code>"sample_id"</code> value.
If any values are missing, they will be filled in using
<code><a href='https://rdrr.io/pkg/colorjam/man/rainbowJam.html'>colorjam::rainbowJam()</a></code>.</p></td>
    </tr>
    <tr>
      <th>ylabel</th>
      <td><p>character string used as the y-axis label, by default
<code>"score"</code> reflects the coverage score and junction score,
respectively for coverage and junction data. Scores are
also adjusted using the <code>scale_factor</code> value for each
<code>sample_id</code> as defined in the <code>filesDF</code>. Set to <code>NULL</code> to
hide the y-axis label completely.</p></td>
    </tr>
    <tr>
      <th>xlabel</th>
      <td><p>character string used to define the x-axis name,
which takes priority over argument <code>xlabel_ref</code>. When
<code>xlabel</code> is <code>NULL</code> and <code>xlabel_ref</code> is <code>FALSE</code>, then the
x-axis name is <code>""</code>, which displays no x-axis label.</p></td>
    </tr>
    <tr>
      <th>xlabel_ref</th>
      <td><p>logical indicating whether the x-axis name
should be determined by the reference (chromosome).</p></td>
    </tr>
    <tr>
      <th>use_jam_themes</th>
      <td><p>logical indicating whether to apply
<code><a href='https://rdrr.io/pkg/colorjam/man/theme_jam.html'>colorjam::theme_jam()</a></code>, by default for the ggplot theme.</p></td>
    </tr>
    <tr>
      <th>apply_facet</th>
      <td><p>logical indicating whether to apply
<code><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>ggplot2::facet_wrap()</a></code> with <code>"~sample_id"</code> defining each
panel.</p></td>
    </tr>
    <tr>
      <th>facet_scales</th>
      <td><p>character value used as <code>"scales"</code> argument in
<code><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>ggplot2::facet_wrap()</a></code> when <code>apply_facet=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>ref2c</th>
      <td><p>optional output from <code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code> used to
compress axis coordinates during junction arc calculations.</p></td>
    </tr>
    <tr>
      <th>label_coords</th>
      <td><p>numeric vector length 2, optional range of
genomic coordinates to restrict labels, so labels are not
arranged by <code><a href='https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html'>ggrepel::geom_text_repel()</a></code> even when <code><a href='https://ggplot2.tidyverse.org/reference/coord_cartesian.html'>coord_cartesian()</a></code>
is used to zoom into a specific x-axis range.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical indicating whether to print verbose output.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments are sent to <code><a href='grl2df.html'>grl2df()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function uses Sashimi data prepared by <code><a href='prepareSashimi.html'>prepareSashimi()</a></code>
and creates a ggplot graphical object ready for visualization.
As a result, this function provides several arguments to
customize the visualization.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other jam plot functions: 
<code><a href='bgaPlotly3d.html'>bgaPlotly3d</a>()</code>,
<code><a href='factor2label.html'>factor2label</a>()</code>,
<code><a href='gene2gg.html'>gene2gg</a>()</code>,
<code><a href='grl2df.html'>grl2df</a>()</code>,
<code><a href='jitter_norm.html'>jitter_norm</a>()</code>,
<code><a href='prepareSashimi.html'>prepareSashimi</a>()</code>,
<code><a href='stackJunctions.html'>stackJunctions</a>()</code></p>
<p>Other jam ggplot2 functions: 
<code><a href='gene2gg.html'>gene2gg</a>()</code>,
<code><a href='geom_diagonal_wide_arc.html'>geom_diagonal_wide_arc</a>()</code>,
<code><a href='splicejam-extensions.html'>splicejam-extensions</a></code>,
<code><a href='to_basic.GeomShape.html'>to_basic.GeomShape</a>()</code></p>
<p>Other splicejam core functions: 
<code><a href='exoncov2polygon.html'>exoncov2polygon</a>()</code>,
<code><a href='gene2gg.html'>gene2gg</a>()</code>,
<code><a href='grl2df.html'>grl2df</a>()</code>,
<code><a href='make_ref2compressed.html'>make_ref2compressed</a>()</code>,
<code><a href='prepareSashimi.html'>prepareSashimi</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/message.html'>suppressPackageStartupMessages</a></span><span class='op'>(</span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>GenomicRanges</span><span class='op'>)</span><span class='op'>)</span>;
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_exon_gr</span><span class='op'>)</span>;
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_junc_gr</span><span class='op'>)</span>;
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_cov_gr</span><span class='op'>)</span>;
<span class='va'>filesDF</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>url<span class='op'>=</span><span class='st'>"sample_A"</span>,
   type<span class='op'>=</span><span class='st'>"coverage_gr"</span>,
   sample_id<span class='op'>=</span><span class='st'>"sample_A"</span><span class='op'>)</span>;
<span class='va'>sh1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='prepareSashimi.html'>prepareSashimi</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList</a></span><span class='op'>(</span>TestGene1<span class='op'>=</span><span class='va'>test_exon_gr</span><span class='op'>)</span>,
   filesDF<span class='op'>=</span><span class='va'>filesDF</span>,
   gene<span class='op'>=</span><span class='st'>"TestGene1"</span>,
   covGR<span class='op'>=</span><span class='va'>test_cov_gr</span>,
   juncGR<span class='op'>=</span><span class='va'>test_junc_gr</span><span class='op'>)</span>;
</div><div class='output co'>#&gt; <span class='warning'>Warning: invalid factor level, NA generated</span></div><div class='output co'>#&gt; <span class='warning'>Warning: invalid factor level, NA generated</span></div><div class='input'><span class='fu'>plotSashimi</span><span class='op'>(</span><span class='va'>sh1</span><span class='op'>)</span>;
</div><div class='img'><img src='plotSashimi-1.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


