<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Prepare Sashimi plot data — prepareSashimi • splicejam</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Prepare Sashimi plot data — prepareSashimi" />
<meta property="og:description" content="Prepare Sashimi plot data" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">splicejam</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.64.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/basic-rnaseq-workflow.html">Splicejam basic RNAseq workflow</a>
    </li>
    <li>
      <a href="../articles/create-a-sashimi-plot.html">Create a Sashimi Plot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/jmw86069/splicejam/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare Sashimi plot data</h1>
    <small class="dont-index">Source: <a href='http://github.com/jmw86069/splicejam/blob/master/R/jambio-plot.R'><code>R/jambio-plot.R</code></a></small>
    <div class="hidden name"><code>prepareSashimi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Prepare Sashimi plot data</p>
    </div>

    <pre class="usage"><span class='fu'>prepareSashimi</span><span class='op'>(</span>
  flatExonsByGene <span class='op'>=</span> <span class='cn'>NULL</span>,
  filesDF <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>gene</span>,
  sample_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  minJunctionScore <span class='op'>=</span> <span class='fl'>10</span>,
  gapWidth <span class='op'>=</span> <span class='fl'>200</span>,
  addGaps <span class='op'>=</span> <span class='cn'>TRUE</span>,
  baseline <span class='op'>=</span> <span class='fl'>0</span>,
  compressGR <span class='op'>=</span> <span class='cn'>TRUE</span>,
  compress_introns <span class='op'>=</span> <span class='cn'>TRUE</span>,
  ref2c <span class='op'>=</span> <span class='cn'>NULL</span>,
  gap_feature_type <span class='op'>=</span> <span class='st'>"intron"</span>,
  default_feature_type <span class='op'>=</span> <span class='st'>"exon"</span>,
  feature_type_colname <span class='op'>=</span> <span class='st'>"feature_type"</span>,
  exon_label_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"none"</span>, <span class='st'>"repel"</span>, <span class='st'>"mark"</span><span class='op'>)</span>,
  junc_label_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"repel"</span>, <span class='st'>"mark"</span>, <span class='st'>"none"</span><span class='op'>)</span>,
  return_data <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"df"</span>, <span class='st'>"ref2c"</span><span class='op'>)</span>,
  include_strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"both"</span>, <span class='st'>"+"</span>, <span class='st'>"-"</span><span class='op'>)</span>,
  junc_color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/jamba/man/alpha2col.html'>alpha2col</a></span><span class='op'>(</span><span class='st'>"goldenrod3"</span>, <span class='fl'>0.7</span><span class='op'>)</span>,
  junc_fill <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/jamba/man/alpha2col.html'>alpha2col</a></span><span class='op'>(</span><span class='st'>"goldenrod1"</span>, <span class='fl'>0.4</span><span class='op'>)</span>,
  doStackJunctions <span class='op'>=</span> <span class='cn'>TRUE</span>,
  coord_method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"coord"</span>, <span class='st'>"scale"</span>, <span class='st'>"none"</span><span class='op'>)</span>,
  scoreFactor <span class='op'>=</span> <span class='fl'>1</span>,
  scoreArcFactor <span class='op'>=</span> <span class='fl'>0.2</span>,
  scoreArcMinimum <span class='op'>=</span> <span class='fl'>100</span>,
  covGR <span class='op'>=</span> <span class='cn'>NULL</span>,
  juncGR <span class='op'>=</span> <span class='cn'>NULL</span>,
  use_memoise <span class='op'>=</span> <span class='cn'>FALSE</span>,
  memoise_coverage_path <span class='op'>=</span> <span class='st'>"coverage_memoise"</span>,
  memoise_junction_path <span class='op'>=</span> <span class='st'>"junctions_memoise"</span>,
  do_shiny_progress <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>flatExonsByGene</th>
      <td><p>GRangesList named by gene, whose GRanges
elements are flattened, disjoint, non-overlapping genomic ranges
per gene.</p></td>
    </tr>
    <tr>
      <th>filesDF</th>
      <td><p>data.frame with columns <code>url</code>, <code>sample_id</code>, <code>type</code>,
where: <code>url</code> is any valid file path or URL compatible with
<code>base::read.table()</code>; <code>sample_id</code> is an identified representing
a biological sample, used to group common files together;
<code>type</code> is one of <code>"bw"</code> for bigWig coverage, <code>"junction"</code> for
BED12 format splice junctions.</p></td>
    </tr>
    <tr>
      <th>gene</th>
      <td><p>character string of the gene to prepare, which must be
present in <code><a href='https://rdrr.io/r/base/names.html'>names(flatExonsByGene)</a></code>.</p></td>
    </tr>
    <tr>
      <th>gapWidth</th>
      <td><p>numeric value of the fixed width to use for
gaps (introns) between exon features. If <code>NULL</code> then
<code><a href='getGRgaps.html'>getGRgaps()</a></code> will use the default based upon the median exon
width.</p></td>
    </tr>
    <tr>
      <th>addGaps</th>
      <td><p>logical indicating whether to include gap regions
in the coverage plot, for example including introns or intergenic
regions. When <code>compressGR=TRUE</code> then gaps regions are
down-sampled using running maximum signal with roughly the same
x-axis resolution as uncompressed regions.</p></td>
    </tr>
    <tr>
      <th>baseline</th>
      <td><p>numeric vector named by <code><a href='https://rdrr.io/r/base/names.html'>names(flatExonsByGene)</a></code>
where baseline is used to adjust the y-axis baseline position
above or below zero.</p></td>
    </tr>
    <tr>
      <th>compressGR</th>
      <td><p>logical indicating whether to compress GRanges
coordinates in the output data, where gaps/introns are set
to a fixed width. When <code>ref2c</code> is not supplied, and
<code>compressGR=TRUE</code>, then <code>ref2c</code> is created using
<code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code>.</p></td>
    </tr>
    <tr>
      <th>compress_introns</th>
      <td><p>logical indicating whether to compress
the coverage polygon coordinates to approximately the same
number of pixels per inch as the exon polygons. This option
greatly reduces the size of the polygon, since introns are
already about 50 to 100 times wider than exons, and when
<code>compressGR</code> is <code>TRUE</code>, the introns are visibly compressed
to a fixed width on the x-axis. The data has many more
x-axis coordinates than the data visualization, this argument
is intended to reduce the intron coordinates accordingly.</p></td>
    </tr>
    <tr>
      <th>ref2c</th>
      <td><p>list object output from <code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code> used
to compress axis coordinates, to compress polygon coverage
data in compressed regions, and to adjust splice junction arcs
using compressed coordinates.</p></td>
    </tr>
    <tr>
      <th>gap_feature_type</th>
      <td><p>the default feature_type value to use for
gaps when <code>addGaps=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>include_strand</th>
      <td><p>character value, one of <code>"both"</code>, <code>"+"</code>,
<code>"-"</code> indicating the strandedness of coverage and junctions
to display. The default <code>"both"</code> shows coverage on both strands,
otherwise coverage is filtered either by filename (presence of
<code>"pos"</code>, <code>"+"</code>, or <code>"plus"</code> indicates positive strand), or
by detecting strandedness by positive/negative coverage scores.
Detecting by filename is intended to avoid retrieving coverage
in the R-shiny app, to help efficiency.</p></td>
    </tr>
    <tr>
      <th>doStackJunctions</th>
      <td><p>logical indicating whether to stack
junction arcs at each end, this argument is passed to
<code><a href='grl2df.html'>grl2df()</a></code> which calls <code><a href='stackJunctions.html'>stackJunctions()</a></code>.</p></td>
    </tr>
    <tr>
      <th>covGR</th>
      <td><p>GRanges object containing coverage data in columns
stored as NumericList class, where <code><a href='https://rdrr.io/r/base/colnames.html'>colnames(values(covGR))</a></code>
are present in <code>filesDF$url</code> when <code>files$type %in% "coverage_gr"</code>.</p></td>
    </tr>
    <tr>
      <th>juncGR</th>
      <td><p>GRanges object containing splice junctions, where
<code>"score"</code> is used for the abundance of splice junction reads,
and <code>"sample_id"</code> is used to define the biological <code>sample_id</code>.</p></td>
    </tr>
    <tr>
      <th>do_shiny_progress</th>
      <td><p>logical indicating whether to send
progress updates to a running shiny app, using the
<code><a href='https://shiny.rstudio.com/reference/shiny/latest/withProgress.html'>shiny::withProgress()</a></code> and <code><a href='https://shiny.rstudio.com/reference/shiny/latest/withProgress.html'>shiny::setProgress()</a></code> methods.
This function only calls <code><a href='https://shiny.rstudio.com/reference/shiny/latest/withProgress.html'>shiny::setProgress()</a></code> and
assumes the <code><a href='https://shiny.rstudio.com/reference/shiny/latest/withProgress.html'>shiny::withProgress()</a></code> has already been
initialized.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical indicating whether to print verbose output.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments are passed to <code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code>,
<code><a href='getGRcoverageFromBw.html'>getGRcoverageFromBw()</a></code>, <code><a href='exoncov2polygon.html'>exoncov2polygon()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list containing <code>ggSashimi</code> a ggplot2 graphical object
containing a full Sashimi plot; <code>ggCov</code> the RNA-seq coverage
subset of the Sashimi plot; <code>ggJunc</code> the splice junction
subsset of the Sashimi plot; <code>ref2c</code> the output of
<code><a href='make_ref2compressed.html'>make_ref2compressed()</a></code> used for ggplot2 coordinate
visualization; <code>covDF</code>, <code>juncDF</code> data.frame objects
with the raw data used to create ggplot2 objects;
<code>covGR</code>, <code>juncGR</code> the GRanges objects used to create
the data.frames; <code>gr</code> the GRanges object representing the
exons for the gene of interest; <code>juncLabelDF</code> the data.frame
containing exon label coordinates used to add labels to
the splice junction arcs.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function is the workhorse function used to produce
Sashimi plots, and is intended to be a convenient wrapper
function for several other individual functions.</p>
<p>At a minimum, a Sashimi plot requires three things:</p><ol>
<li><p>Exons, usually from a gene of interest.</p></li>
<li><p>RNA-seq coverage data.</p></li>
<li><p>Splice junction data.</p></li>
</ol>

<p>There is some required pre-processing before running
<code>prepareSashimi()</code>:</p><ul>
<li><p>Prepare flattened exons by gene using <code>flattenExonsByGene()</code>
and corresponding data, including <code>exonsByGene</code>, <code>cdsByGene</code>,
and <code>tx2geneDF</code>. Verify the gene exon model data using
<code><a href='gene2gg.html'>gene2gg()</a></code>.</p></li>
<li><p>Find file paths, or web URLs, for a set of bigWig coverage
files, representing RNA-seq coverage for each strand, for
the samples of interest. Test the coverage data using
<code><a href='getGRcoverageFromBw.html'>getGRcoverageFromBw()</a></code> for a small set of GRanges data.</p></li>
<li><p>Find file paths, or web URLs, for a set of BED6 or BED12
format files, note that it cannot currently use bigBed format
due to limitations in the <code>rtracklayer</code> package.
Test the splice junction data using <code><a href='https://rdrr.io/pkg/rtracklayer/man/export.html'>rtracklayer::import()</a></code>
for a small range of GRanges features, then send the data
to <code><a href='spliceGR2junctionDF.html'>spliceGR2junctionDF()</a></code> to prepare a data.frame summary.</p></li>
</ul>

<p>The basic input for coverage and junction data is a data.frame,
which defines each file path or url, the type of data
<code>"bw"</code> or <code>"junction"</code>, and the biological sample <code>"sample_id"</code>.
Any file path compatible with <code><a href='https://rdrr.io/pkg/rtracklayer/man/export.html'>rtracklayer::import()</a></code> will
work, including web URLs and local files. When using a web URL
you may need to use <code>"https://"</code> format to force the use
of secure web requests, but this requirement varies by country.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other jam RNA-seq functions: 
<code><a href='assignGRLexonNames.html'>assignGRLexonNames</a>()</code>,
<code><a href='closestExonToJunctions.html'>closestExonToJunctions</a>()</code>,
<code><a href='combineGRcoverage.html'>combineGRcoverage</a>()</code>,
<code><a href='defineDetectedTx.html'>defineDetectedTx</a>()</code>,
<code><a href='detectedTxInfo.html'>detectedTxInfo</a>()</code>,
<code><a href='exoncov2polygon.html'>exoncov2polygon</a>()</code>,
<code><a href='flattenExonsBy.html'>flattenExonsBy</a>()</code>,
<code><a href='getGRcoverageFromBw.html'>getGRcoverageFromBw</a>()</code>,
<code><a href='groups2contrasts.html'>groups2contrasts</a>()</code>,
<code><a href='internal_junc_score.html'>internal_junc_score</a>()</code>,
<code><a href='makeTx2geneFromGtf.html'>makeTx2geneFromGtf</a>()</code>,
<code><a href='make_ref2compressed.html'>make_ref2compressed</a>()</code>,
<code><a href='runDiffSplice.html'>runDiffSplice</a>()</code>,
<code><a href='sortSamples.html'>sortSamples</a>()</code>,
<code><a href='spliceGR2junctionDF.html'>spliceGR2junctionDF</a>()</code></p>
<p>Other jam plot functions: 
<code><a href='bgaPlotly3d.html'>bgaPlotly3d</a>()</code>,
<code><a href='factor2label.html'>factor2label</a>()</code>,
<code><a href='gene2gg.html'>gene2gg</a>()</code>,
<code><a href='grl2df.html'>grl2df</a>()</code>,
<code><a href='jitter_norm.html'>jitter_norm</a>()</code>,
<code><a href='plotSashimi.html'>plotSashimi</a>()</code>,
<code><a href='stackJunctions.html'>stackJunctions</a>()</code></p>
<p>Other splicejam core functions: 
<code><a href='exoncov2polygon.html'>exoncov2polygon</a>()</code>,
<code><a href='gene2gg.html'>gene2gg</a>()</code>,
<code><a href='grl2df.html'>grl2df</a>()</code>,
<code><a href='make_ref2compressed.html'>make_ref2compressed</a>()</code>,
<code><a href='plotSashimi.html'>plotSashimi</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># The active example below uses sample data</span>
<span class='fu'><a href='https://rdrr.io/r/base/message.html'>suppressPackageStartupMessages</a></span><span class='op'>(</span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>GenomicRanges</span><span class='op'>)</span><span class='op'>)</span>;

<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_exon_gr</span><span class='op'>)</span>;
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_junc_gr</span><span class='op'>)</span>;
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>test_cov_gr</span><span class='op'>)</span>;
<span class='va'>filesDF</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>url<span class='op'>=</span><span class='st'>"sample_A"</span>,
   type<span class='op'>=</span><span class='st'>"coverage_gr"</span>,
   sample_id<span class='op'>=</span><span class='st'>"sample_A"</span><span class='op'>)</span>;
<span class='va'>sh1</span> <span class='op'>&lt;-</span> <span class='fu'>prepareSashimi</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList</a></span><span class='op'>(</span>TestGene1<span class='op'>=</span><span class='va'>test_exon_gr</span><span class='op'>)</span>,
   filesDF<span class='op'>=</span><span class='va'>filesDF</span>,
   gene<span class='op'>=</span><span class='st'>"TestGene1"</span>,
   covGR<span class='op'>=</span><span class='va'>test_cov_gr</span>,
   juncGR<span class='op'>=</span><span class='va'>test_junc_gr</span><span class='op'>)</span>;
</div><div class='output co'>#&gt; <span class='warning'>Warning: invalid factor level, NA generated</span></div><div class='output co'>#&gt; <span class='warning'>Warning: invalid factor level, NA generated</span></div><div class='input'><span class='fu'><a href='plotSashimi.html'>plotSashimi</a></span><span class='op'>(</span><span class='va'>sh1</span><span class='op'>)</span>;
</div><div class='img'><img src='prepareSashimi-1.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


