<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform differential isoform analysis using diffSplice — runDiffSplice • splicejam</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Perform differential isoform analysis using diffSplice — runDiffSplice" />
<meta property="og:description" content="Perform differential isoform analysis using diffSplice" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">splicejam</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.63.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/basic-rnaseq-workflow.html">Splicejam basic RNAseq workflow</a>
    </li>
    <li>
      <a href="../articles/create-a-sashimi-plot.html">Create a Sashimi Plot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/jmw86069/splicejam/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform differential isoform analysis using diffSplice</h1>
    <small class="dont-index">Source: <a href='http://github.com/jmw86069/splicejam/blob/master/R/jambio-tx.R'><code>R/jambio-tx.R</code></a></small>
    <div class="hidden name"><code>runDiffSplice.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform differential isoform analysis using diffSplice</p>
    </div>

    <pre class="usage"><span class='fu'>runDiffSplice</span><span class='op'>(</span>
  <span class='va'>iMatrixTx</span>,
  detectedTx <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>iMatrixTx</span><span class='op'>)</span>,
  <span class='va'>tx2geneDF</span>,
  txColname <span class='op'>=</span> <span class='st'>"transcript_id"</span>,
  geneColname <span class='op'>=</span> <span class='st'>"gene_name"</span>,
  <span class='va'>iDesign</span>,
  <span class='va'>iContrasts</span>,
  cutoffFDR <span class='op'>=</span> <span class='fl'>0.05</span>,
  cutoffFold <span class='op'>=</span> <span class='fl'>1.5</span>,
  collapseByGene <span class='op'>=</span> <span class='cn'>TRUE</span>,
  spliceTest <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"t"</span>, <span class='st'>"simes"</span>, <span class='st'>"F"</span><span class='op'>)</span>,
  sep <span class='op'>=</span> <span class='st'>" "</span>,
  useVoom <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>iMatrixTx</th>
      <td><p>numeric matrix of expression, with transcripts as
rows and samples as columns. The data is assumed to be log2-transformed
using the format <code><a href='https://rdrr.io/r/base/Log.html'>log2(1 + x)</a></code>. This data should be normalized using
appropriate methods, outside the scope of this function.</p></td>
    </tr>
    <tr>
      <th>detectedTx</th>
      <td><p>character vector containing all or a subset of
<code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iMatrix)</a></code> used for statistical testing.</p></td>
    </tr>
    <tr>
      <th>tx2geneDF</th>
      <td><p>data.frame with colnames <code><a href='https://rdrr.io/r/base/c.html'>c(txColname, geneColname)</a></code>,
where all entries of <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iMatrix)</a></code> are represented
in <code>tx2geneDF[,txColname]</code>.</p></td>
    </tr>
    <tr>
      <th>txColname, geneColname</th>
      <td><p>the <code><a href='https://rdrr.io/r/base/colnames.html'>colnames(tx2geneDF)</a></code> representing
the <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iMatrixTx)</a></code> matched by <code>tx2geneDF[,txColname]</code>,
and the associated genes given by <code>tx2geneDF[,geneColname]</code>.
Note that <code>detectedTx</code> must also contain values in <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iMatrixTx)</a></code>
and <code>tx2geneDF[,txColname]</code>.</p></td>
    </tr>
    <tr>
      <th>iDesign</th>
      <td><p>numeric matrix representing the design matrix for
the experiment design. For example, <code>limma::model.matrix(~0+group)</code>
will represent each group. Typically, <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iDesign)</a></code> should
be defined to match the <code><a href='https://rdrr.io/r/base/colnames.html'>colnames(iMatrix)</a></code> even if it requires
extra processing. The <code><a href='https://rdrr.io/r/base/colnames.html'>colnames(iDesign)</a></code> should represent
group names used in <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iContrasts)</a></code>.</p></td>
    </tr>
    <tr>
      <th>iContrasts</th>
      <td><p>numeric matrix representing the contrasts used
in statistical comparisons. This matrix can be generated by
running <code><a href='https://rdrr.io/pkg/limma/man/makeContrasts.html'>limma::makeContrasts()</a></code> using a format similar to the
following: <code><a href='https://rdrr.io/pkg/limma/man/makeContrasts.html'>limma::makeContrasts(contrasts="group1-group2", levels=iDesign)</a></code>,
see "Examples" for more info.</p></td>
    </tr>
    <tr>
      <th>cutoffFDR</th>
      <td><p>numeric value indicating a statistical threshold
on the FDR (adjusted P-value). Values should be between 0 and 1, where
<code>cutoffFDR=1</code> would impose no threshold on the adjusted P-value.</p></td>
    </tr>
    <tr>
      <th>cutoffFold</th>
      <td><p>numeric value indicating the minimum normal space
fold change allowed for statistical hits. For example <code>cutoffFold=2</code>
would require a 2-fold change, equivalent to log2 fold change &gt;= 1.</p></td>
    </tr>
    <tr>
      <th>collapseByGene</th>
      <td><p>logical indicating whether results should be
summarized at the gene level after filtering statistical hits.</p></td>
    </tr>
    <tr>
      <th>spliceTest</th>
      <td><p>character value described in <code><a href='https://rdrr.io/pkg/limma/man/topSplice.html'>limma::topSplice()</a></code>
which defines the statistical test to return. The default <code>"t"</code>
returns the t-test result for each isoform, which is mainly beneficial
because it also includes fold change that can be filtered. The
<code>"F"</code> returns F-test per gene, and <code>"simes"</code> returns the per-gene
t-test P-value after Simes adjustment per gene.</p></td>
    </tr>
    <tr>
      <th>sep</th>
      <td><p>character value used as a delimiter in output data.frame
colnames, such that each stats is followed by the contrast name,
separated by this delimiter.</p></td>
    </tr>
    <tr>
      <th>useVoom</th>
      <td><p>logical indicating whether to apply the <code><a href='https://rdrr.io/pkg/limma/man/voom.html'>limma::voom()</a></code>
adjustment prior to running <code><a href='https://rdrr.io/pkg/limma/man/diffSplice.html'>limma::diffSplice()</a></code>. This value
should be <code>TRUE</code> when analyzing count or pseudocount data.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical indicating whether to print verbose output.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments are ignored.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list containing:</p><ul>
<li><p><code>detectedTx</code>, <code>detectedTxUse</code> the detectedTx
values representing genes with multiple transcripts;</p></li>
<li><p><code>fit</code> the initial model fit;</p></li>
<li><p><code>fit2</code> the contrast model fit;</p></li>
<li><p><code>splice</code> the output from <code><a href='https://rdrr.io/pkg/limma/man/diffSplice.html'>limma::diffSplice()</a></code>;</p></li>
<li><p><code>statsDFs</code> list of data.frame output from <code><a href='https://rdrr.io/pkg/limma/man/topSplice.html'>limma::topSplice()</a></code>
either at the transcript level or the gene level.</p></li>
</ul>

<p>Note that when <code>collapseByGene=TRUE</code> the results will return the first
transcript entry per gene that has the best P-value. Often one gene
will have two transcripts with identical P-value, and the order
that the transcripts appear is inconsistent. Therefore, the direction
of fold change is not meaningful by itself, except with respect to
the specific isoform returned. See <code><a href='https://rdrr.io/pkg/limma/man/topSplice.html'>limma::topSplice()</a></code> argument
<code>test</code> for more information about transcript- and gene-level
summaries.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function is intended to be a convenient method
to call <code><a href='https://rdrr.io/pkg/limma/man/diffSplice.html'>limma::diffSplice()</a></code> and return the output of
<code><a href='https://rdrr.io/pkg/limma/man/topSplice.html'>limma::topSplice()</a></code> in helpful formats for downstream
use.</p>
<p>The basic input required:</p><ul>
<li><p><code>iMatrixTx</code> a numeric matrix of transcript expression</p></li>
<li><p><code>detectedTx</code> (optional) subset of <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(iMatrixTx)</a></code>, sometimes
determined by <code><a href='defineDetectedTx.html'>defineDetectedTx()</a></code>.</p></li>
<li><p><code>tx2geneDF</code> data.frame with "transcript_id" and "gene_name" columns.
This data.frame is often the same one used with <code><a href='https://rdrr.io/pkg/tximport/man/tximport.html'>tximport::tximport()</a></code>
when importing transcriptome data to the gene level.</p></li>
<li><p><code>iDesign</code>,<code>iContrasts</code> design and contrast matrix, as created by
<code><a href='groups2contrasts.html'>groups2contrasts()</a></code>.</p></li>
</ul>

<p>These steps are run in order:</p><ul>
<li><p><code>limma:voom()</code> (Optional.) This step is enabled with
the argument <code>useVoom=TRUE</code> and should only be used when <code>iMatrixTx</code>
contains count or pseudocount data. When using TPM or FPKM values,
set <code>useVoom=FALSE</code>.</p></li>
<li><p><code><a href='https://rdrr.io/pkg/limma/man/lmFit.html'>limma::lmFit()</a></code></p></li>
<li><p><code><a href='https://rdrr.io/pkg/limma/man/contrasts.fit.html'>limma::contrasts.fit()</a></code></p></li>
<li><p><code><a href='https://rdrr.io/pkg/limma/man/diffSplice.html'>limma::diffSplice()</a></code></p></li>
<li><p><code><a href='https://rdrr.io/pkg/limma/man/topSplice.html'>limma::topSplice()</a></code> This function is called on each contrast
in order to return a list of data.frames.</p></li>
</ul>

<p>Each contrast is tested for differential transcript expression.
No other contrasts are tested.</p>
<p>The output is a list with an element <code>"statsDFs"</code> that is itself
list of data.frames for each contrast. By default when
<code>collapseByGene=TRUE</code> each row is collapsed to gene level,
using the best statistical hit per gene as an exemplar.</p>
<p>The rows  of <code>iMatrixTx</code> are expected to contain expression values
per transcript isoform, but may contain alternative measurements
such as: junction counts per gene; exon expression per gene.</p>
<p>When <code>useVoom=TRUE</code>, the <code>iMatrixTx</code> data is exponentiated
prior to running <code><a href='https://rdrr.io/pkg/limma/man/voom.html'>limma::voom()</a></code>, for the purpose of
calculating a weights matrix.
The original <code>iMatrixTx</code> data is used in <code><a href='https://rdrr.io/pkg/limma/man/lmFit.html'>limma::lmFit()</a></code> as-is,
alongside the voom weights. The voom-normalized data is not
used. Therefore, the input data is assumed to be normalized.</p>
<p>Statistical results can be summarized at the gene level, after
applying thresholds for statistical hits in the form of
required adjusted P-value and/or fold change. It may be helpful
to review results per gene, alongside the specific transcript
isoforms which are called statistical hits.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Law, CW, Chen, Y, Shi, W, and Smyth, GK (2014).
Voom: precision weights unlock linear model analysis
tools for RNA-seq read counts.
<em>Genome Biology</em> <strong>15</strong>, R29.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other jam RNA-seq functions: 
<code><a href='assignGRLexonNames.html'>assignGRLexonNames</a>()</code>,
<code><a href='closestExonToJunctions.html'>closestExonToJunctions</a>()</code>,
<code><a href='combineGRcoverage.html'>combineGRcoverage</a>()</code>,
<code><a href='defineDetectedTx.html'>defineDetectedTx</a>()</code>,
<code><a href='detectedTxInfo.html'>detectedTxInfo</a>()</code>,
<code><a href='exoncov2polygon.html'>exoncov2polygon</a>()</code>,
<code><a href='flattenExonsBy.html'>flattenExonsBy</a>()</code>,
<code><a href='groups2contrasts.html'>groups2contrasts</a>()</code>,
<code><a href='makeTx2geneFromGtf.html'>makeTx2geneFromGtf</a>()</code>,
<code><a href='make_ref2compressed.html'>make_ref2compressed</a>()</code>,
<code><a href='sortSamples.html'>sortSamples</a>()</code>,
<code><a href='spliceGR2junctionDF.html'>spliceGR2junctionDF</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># example for defining iDesign</span>
<span class='co'># first define a vector of sample groups</span>
<span class='va'>iGroups</span> <span class='op'>&lt;-</span> <span class='fu'>jamba</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/jamba/man/nameVector.html'>nameVector</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"WT"</span>, <span class='st'>"KO"</span><span class='op'>)</span>, each<span class='op'>=</span><span class='fl'>6</span><span class='op'>)</span>,
   <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Control"</span>, <span class='st'>"Treated"</span><span class='op'>)</span>, each<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span>,
   sep<span class='op'>=</span><span class='st'>"_"</span><span class='op'>)</span><span class='op'>)</span>;
<span class='va'>iGroups</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>iGroups</span>, levels<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>iGroups</span><span class='op'>)</span><span class='op'>)</span>;
<span class='va'>iGroups</span>;
</div><div class='output co'>#&gt; WT_Control_v1 WT_Control_v2 WT_Control_v3 WT_Treated_v1 WT_Treated_v2 
#&gt;    WT_Control    WT_Control    WT_Control    WT_Treated    WT_Treated 
#&gt; WT_Treated_v3 KO_Control_v1 KO_Control_v2 KO_Control_v3 KO_Treated_v1 
#&gt;    WT_Treated    KO_Control    KO_Control    KO_Control    KO_Treated 
#&gt; KO_Treated_v2 KO_Treated_v3 
#&gt;    KO_Treated    KO_Treated 
#&gt; Levels: WT_Control WT_Treated KO_Control KO_Treated</div><div class='input'><span class='co'># next define sample identifiers</span>
<span class='va'>iSamples</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>iGroups</span><span class='op'>)</span>;

<span class='co'># given a vector of groups, make iDesign</span>
<span class='va'>iDesign</span> <span class='op'>&lt;-</span> <span class='fu'>stats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/r/stats/model.matrix.html'>model.matrix</a></span><span class='op'>(</span><span class='op'>~</span><span class='fl'>0</span><span class='op'>+</span><span class='va'>iGroups</span><span class='op'>)</span>;

<span class='co'># It is good practice to rename colnames(iDesign) and rownames(iDesign),</span>
<span class='co'># that is, for the love of all that is good, use colnames and rownames</span>
<span class='co'># that help confirm that these matrices are consistent.</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>iDesign</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>iGroups</span><span class='op'>)</span>;
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>iDesign</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>iGroups</span><span class='op'>)</span>;
<span class='va'>iDesign</span>;
</div><div class='output co'>#&gt;               WT_Control WT_Treated KO_Control KO_Treated
#&gt; WT_Control_v1          1          0          0          0
#&gt; WT_Control_v2          1          0          0          0
#&gt; WT_Control_v3          1          0          0          0
#&gt; WT_Treated_v1          0          1          0          0
#&gt; WT_Treated_v2          0          1          0          0
#&gt; WT_Treated_v3          0          1          0          0
#&gt; KO_Control_v1          0          0          1          0
#&gt; KO_Control_v2          0          0          1          0
#&gt; KO_Control_v3          0          0          1          0
#&gt; KO_Treated_v1          0          0          0          1
#&gt; KO_Treated_v2          0          0          0          1
#&gt; KO_Treated_v3          0          0          0          1
#&gt; attr(,"assign")
#&gt; [1] 1 1 1 1
#&gt; attr(,"contrasts")
#&gt; attr(,"contrasts")$iGroups
#&gt; [1] "contr.treatment"
#&gt; </div><div class='input'>
<span class='co'># define contrasts</span>
<span class='co'># the example below includes a two-way contrast, which is a test</span>
<span class='co'># of the pairwise fold changes</span>
<span class='va'>iContrasts</span> <span class='op'>&lt;-</span> <span class='fu'>limma</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/makeContrasts.html'>makeContrasts</a></span><span class='op'>(</span>contrasts<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
   <span class='st'>"WT_Treated-WT_Control"</span>, <span class='st'>"KO_Treated-KO_Control"</span>,
   <span class='st'>"(KO_Treated-KO_Control)-(WT_Treated-WT_Control)"</span><span class='op'>)</span>,
   levels<span class='op'>=</span><span class='va'>iDesign</span><span class='op'>)</span>;
<span class='va'>iContrasts</span>;
</div><div class='output co'>#&gt;             Contrasts
#&gt; Levels       WT_Treated-WT_Control KO_Treated-KO_Control
#&gt;   WT_Control                    -1                     0
#&gt;   WT_Treated                     1                     0
#&gt;   KO_Control                     0                    -1
#&gt;   KO_Treated                     0                     1
#&gt;             Contrasts
#&gt; Levels       (KO_Treated-KO_Control)-(WT_Treated-WT_Control)
#&gt;   WT_Control                                               1
#&gt;   WT_Treated                                              -1
#&gt;   KO_Control                                              -1
#&gt;   KO_Treated                                               1</div><div class='input'>
<span class='co'># for validation, verify these constraints:</span>
<span class='co'># - all(rownames(iDesign) == iSamples)</span>
<span class='co'># - colnames(iDesign) == the actual group names</span>
<span class='co'># - all(colnames(iDesign) == rownames(iContrasts))</span>
<span class='co'># you can see which samples are included in each test with crossproduct:</span>
<span class='va'>iDesign</span> <span class='op'>%*%</span> <span class='va'>iContrasts</span>;
</div><div class='output co'>#&gt;                Contrasts
#&gt;                 WT_Treated-WT_Control KO_Treated-KO_Control
#&gt;   WT_Control_v1                    -1                     0
#&gt;   WT_Control_v2                    -1                     0
#&gt;   WT_Control_v3                    -1                     0
#&gt;   WT_Treated_v1                     1                     0
#&gt;   WT_Treated_v2                     1                     0
#&gt;   WT_Treated_v3                     1                     0
#&gt;   KO_Control_v1                     0                    -1
#&gt;   KO_Control_v2                     0                    -1
#&gt;   KO_Control_v3                     0                    -1
#&gt;   KO_Treated_v1                     0                     1
#&gt;   KO_Treated_v2                     0                     1
#&gt;   KO_Treated_v3                     0                     1
#&gt;                Contrasts
#&gt;                 (KO_Treated-KO_Control)-(WT_Treated-WT_Control)
#&gt;   WT_Control_v1                                               1
#&gt;   WT_Control_v2                                               1
#&gt;   WT_Control_v3                                               1
#&gt;   WT_Treated_v1                                              -1
#&gt;   WT_Treated_v2                                              -1
#&gt;   WT_Treated_v3                                              -1
#&gt;   KO_Control_v1                                              -1
#&gt;   KO_Control_v2                                              -1
#&gt;   KO_Control_v3                                              -1
#&gt;   KO_Treated_v1                                               1
#&gt;   KO_Treated_v2                                               1
#&gt;   KO_Treated_v3                                               1</div><div class='input'>
<span class='co'>## Another efficient way to define iDesign and iContrasts:</span>
<span class='va'>iDC</span> <span class='op'>&lt;-</span> <span class='fu'>splicejam</span><span class='fu'>::</span><span class='fu'><a href='groups2contrasts.html'>groups2contrasts</a></span><span class='op'>(</span><span class='va'>iGroups</span>, returnDesign<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>;
<span class='va'>iDesign</span> <span class='op'>&lt;-</span> <span class='va'>iDC</span><span class='op'>$</span><span class='va'>iDesign</span>;
<span class='va'>iContrasts</span> <span class='op'>&lt;-</span> <span class='va'>iDC</span><span class='op'>$</span><span class='va'>iContrasts</span>;

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


