<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Sashimi Shiny app constants — sashimiAppConstants • splicejam</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Sashimi Shiny app constants — sashimiAppConstants" />
<meta property="og:description" content="Sashimi Shiny app constants" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">splicejam</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.63.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/basic-rnaseq-workflow.html">Splicejam basic RNAseq workflow</a>
    </li>
    <li>
      <a href="../articles/create-a-sashimi-plot.html">Create a Sashimi Plot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/jmw86069/splicejam/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Sashimi Shiny app constants</h1>
    <small class="dont-index">Source: <a href='http://github.com/jmw86069/splicejam/blob/master/R/splicejam-shiny-constants.R'><code>R/splicejam-shiny-constants.R</code></a></small>
    <div class="hidden name"><code>sashimiAppConstants.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Sashimi Shiny app constants</p>
    </div>

    <pre class="usage"><span class='fu'>sashimiAppConstants</span><span class='op'>(</span>
  <span class='va'>...</span>,
  assign_global <span class='op'>=</span> <span class='cn'>TRUE</span>,
  empty_uses_farrisdata <span class='op'>=</span> <span class='cn'>TRUE</span>,
  use_memoise <span class='op'>=</span> <span class='cn'>TRUE</span>,
  aboutExtra <span class='op'>=</span> <span class='cn'>NULL</span>,
  filesDF <span class='op'>=</span> <span class='cn'>NULL</span>,
  color_sub <span class='op'>=</span> <span class='cn'>NULL</span>,
  gtf <span class='op'>=</span> <span class='cn'>NULL</span>,
  txdb <span class='op'>=</span> <span class='cn'>NULL</span>,
  tx2geneDF <span class='op'>=</span> <span class='cn'>NULL</span>,
  exonsByTx <span class='op'>=</span> <span class='cn'>NULL</span>,
  cdsByTx <span class='op'>=</span> <span class='cn'>NULL</span>,
  detectedTx <span class='op'>=</span> <span class='cn'>NULL</span>,
  detectedGenes <span class='op'>=</span> <span class='cn'>NULL</span>,
  flatExonsByGene <span class='op'>=</span> <span class='cn'>NULL</span>,
  flatExonsByTx <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>...</th>
      <td><p>additional arguments are ignored.</p></td>
    </tr>
    <tr>
      <th>empty_uses_farrisdata</th>
      <td><p>logical indicating whether to
use data from the <code>"farrisdata"</code> R package if no default
values are provided, and if that package is available.</p></td>
    </tr>
    <tr>
      <th>aboutExtra</th>
      <td><p>character string or html tag from <code>"htmltools"</code>
suitable for use in a shiny app. This text is intended
to display a description of data visualized in the R-shiny
app, for example when run with <code><a href='launchSashimiApp.html'>launchSashimiApp()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function defines several constant values
used by the R-shiny Splicejam Sashimi viewer,
typically taking from the global environment
where possible.</p>
<p>The R-shiny app is started by <code><a href='launchSashimiApp.html'>launchSashimiApp()</a></code>, which
calls <code><a href='https://shiny.rstudio.com/reference/shiny/latest/shinyApp.html'>shiny::shinyApp()</a></code>, using arguments <code>server</code>, <code>ui</code>,
<code>onStart</code>, and <code>options</code>. This function fulfills the
argument <code>onStart</code>.</p>
<p>This function is intended to be called only from within
an R-shiny app, since it defines several variables in the
parent (global) environment.</p>
<p>The following values will be used from the environment,
searching up the environment parent chain until it finds
a match, until searching the global environment. Similarly,
this function also defines variables in the environment using
the <code>&lt;&lt;-</code> operator, which by default also searches up the
environment chain until it finds a match, otherwise
populating the global environment.</p>
<p>If a variable is not found, the corresponding default
from the <code>farrisdata</code> package will be used.</p><ul>
<li><p><strong>filesDF</strong>: <code>data.frame</code> with colnames: <code>"sample_id"</code>,
<code>"type"</code>, <code>"url"</code>, <code>"scale_factor"</code>. When <code>"filesDF"</code> does
not exist, it is taken from <code><a href='https://rdrr.io/pkg/farrisdata/man/farris_sashimi_files_df.html'>farrisdata::farris_sashimi_files_df</a></code>,
and the default <code>"aboutExtra"</code> text is used.</p></li>
<li><p><strong>color_sub</strong>: character vector of R colors, whose names are
used to match <code>"sample_id"</code> or other values used to colorize
ggplot visualizations. It is either populated from
<code><a href='https://rdrr.io/pkg/farrisdata/man/colorSub.html'>farrisdata::colorSub</a></code>, or <code><a href='https://rdrr.io/pkg/colorjam/man/group2colors.html'>colorjam::group2colors()</a></code> is called
using the unique <code>sample_id</code> values in <code>filesDF</code>.</p></li>
<li><p><strong>txdb</strong>: TxDb object used to derive <code>exonsByTx</code>, and <code>cdsByTx</code>
if either does not exist. If the TxDb object does not exist,
it is created from the <code>gtf</code> file using
<code><a href='https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html'>GenomicFeatures::makeTxDbFromGFF()</a></code>.</p></li>
<li><p><strong>tx2geneDF</strong>: <code>data.frame</code> with colnames: <code>"transcript_id"</code>,
<code>"gene_name"</code>. This data is used to convert <code>"gene_name"</code> to
a vector of <code>"transcript_id"</code> values.</p></li>
<li><p><strong>gtf</strong>: character path to a GTF/GFF/GFF3 file, suitable for
<code><a href='https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html'>GenomicFeatures::makeTxDbFromGFF()</a></code>. The <code>gtf</code> is only used
if <code>tx2geneDF</code> or <code>exonsByTx</code> are not available. Note that
when <code>gtf</code> points to a remote server, the file is copied to
the current working directory. If the file already exists in
the local directory, it is re-used.</p></li>
<li><p><strong>exonsByTx</strong>: <code>GRangesList</code> object, named by <code>"transcript_id"</code>,
containing all exons for each transcript.</p></li>
<li><p><strong>cdsByTx</strong>: <code>GRangesList</code> object, named by <code>"transcript_id"</code>,
containing only CDS (protein-coding) exons for each transcript.</p></li>
<li><p><strong>detectedTx</strong>: character vector of <code>"transcript_id"</code> values,
representing the subset of transcripts detected above background.
If it does not exist, it is derived from <code><a href='https://rdrr.io/pkg/farrisdata/man/farrisTxSE.html'>farrisdata::farrisTxSE</a></code>
<code>"TxDetectedByTPM"</code>. If those values are not all present in
<code>tx2geneDF</code> then <code>detectedTx</code> is defined by all transcripts
present in <code>tx2geneDF$transcript_id</code>.</p></li>
<li><p><strong>detectedGenes</strong>: inferred using tx2geneDF and detectedTx.</p></li>
<li><p><strong>flatExonsByGene</strong>: <code>GRangesList</code> object containing non-overlapping
exons for each gene, or it is derived from <code>exonsByTx</code>,
<code>cdsByTx</code>, <code>detectedTx</code>, and <code>tx2geneDF</code>, using
<code><a href='flattenExonsBy.html'>flattenExonsBy()</a></code>.</p></li>
<li><p><strong>flatExonsByTx</strong>: <code>GRangesList</code> object containing non-overlapping
exons for each gene, or it is derived from <code>exonsByTx</code>,
<code>cdsByTx</code>, <code>detectedTx</code>, and <code>tx2geneDF</code>, using
<code><a href='flattenExonsBy.html'>flattenExonsBy()</a></code>.</p></li>
</ul>

<p>Several R objects are cached using <code><a href='https://rdrr.io/pkg/memoise/man/memoise.html'>memoise::memoise()</a></code>, to
avoid having to create the R object each time the R-shiny app
is started:</p><ul>
<li><p><strong>flatExonsByGene</strong></p></li>
<li><p><strong>flatExonsByTx</strong></p></li>
<li><p><strong>exonsByTx</strong></p></li>
<li><p><strong>cdsByTx</strong></p></li>
</ul>

<p>A special custom variable is used to describe the specific
data included with the R-shiny app, <code>"aboutExtra"</code>. The
<code>"aboutExtra"</code> variable is expected to contain HTML tags,
for example from <code><a href='https://rdrr.io/pkg/htmltools/man/builder.html'>htmltools::tags()</a></code> to format the text.
The <code>"aboutExtra"</code> content is displayed on the tab
<code>"About Sashimi Plots"</code> at the top.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other splicejam R-shiny functions: 
<code><a href='launchSashimiApp.html'>launchSashimiApp</a>()</code>,
<code><a href='sashimiAppServer.html'>sashimiAppServer</a>()</code>,
<code><a href='sashimiAppUI.html'>sashimiAppUI</a>()</code></p></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward, Shannon Farris.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


